hack/local-monitor-testing/deploy_MDM_VM.sh

```diff
@@ -2,90 +2,83 @@
+echo "Found IP $MDM_VM_IP"
 
-PUBLICIP=$( az vm list-ip-addresses --name $VMName -g $RESOURCEGROUP | jq -r '.[0].virtualMachine.network.publicIpAddresses[0].ipAddress' )
-
-echo "Found IP $PUBLICIP"
-
-scp $BASE/secrets/rp-metrics-int.pem $CLOUDUSER@$PUBLICIP:mdm.pem
-scp $BASE/hack/local-monitor-testing/configureRemote.sh $CLOUDUSER@$PUBLICIP:
-
-ssh $CLOUDUSER@$PUBLICIP "sudo cp mdm.pem /etc/mdm.pem"
-ssh $CLOUDUSER@$PUBLICIP "sudo ./configureRemote.sh"
+scp -i ./secrets/mdm_id_rsa $BASE/secrets/rp-metrics-int.pem $CLOUDUSER@$MDM_VM_IP:mdm.pem
+scp -i ./secrets/mdm_id_rsa $BASE/hack/local-monitor-testing/configureRemote.sh $CLOUDUSER@$MDM_VM_IP:
 
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo cp mdm.pem /etc/mdm.pem"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo ./configureRemote.sh"
 
-ssh $CLOUDUSER@$PUBLICIP "sudo docker pull $MDMIMAGE"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo docker pull $MDM_IMAGE"
 
 cat <<EOF > $BASE/dockerStartCommand.sh
 docker run \
   --entrypoint /usr/sbin/MetricsExtension \
   --hostname $HOSTNAME \
-  --name $NAME \
+  --name $MDM_CONTAINER_NAME \
   -d \
   --restart=always \
   -m 2g \
   -v /etc/mdm.pem:/etc/mdm.pem \
   -v /var/etw:/var/etw:z \
-  $MDMIMAGE \
+  $MDM_IMAGE \
   -CertFile /etc/mdm.pem \
-  -FrontEndUrl $MDMFRONTENDURL \
+  -FrontEndUrl $MDM_FRONTEND_URL \
   -Logger Console \
   -LogLevel Debug \
   -PrivateKeyFile /etc/mdm.pem \
-  -SourceEnvironment $MDMSOURCEENVIRONMENT \
-  -SourceRole $MDMSOURCEROLE \
-  -SourceRoleInstance $MDMSOURCEROLEINSTANCE
+  -SourceEnvironment $MDM_SOURCE_ENVIRONMENT \
+  -SourceRole $MDM_SOURCE_ROLE \
+  -SourceRoleInstance $MDM_SOURCE_ROLE_INSTANCE
 EOF
 
-
 #disable SELINUX (don't shoot me)
-ssh $CLOUDUSER@$PUBLICIP "sudo setenforce 0"
-ssh $CLOUDUSER@$PUBLICIP "sudo getenforce"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo setenforce 0"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo getenforce"
 
 #make it permanent
-ssh $CLOUDUSER@$PUBLICIP "sudo sed -i 's/SELINUX=enforcing/SELINUX=permissive/g' /etc/selinux/config"
-
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo sed -i 's/SELINUX=enforcing/SELINUX=permissive/g' /etc/selinux/config"
 
-ssh $CLOUDUSER@$PUBLICIP "sudo firewall-cmd --zone=public --add-port=12345/tcp --permanent"
-ssh $CLOUDUSER@$PUBLICIP "sudo firewall-cmd --reload"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo firewall-cmd --zone=public --add-port=12345/tcp --permanent"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo firewall-cmd --reload"
 
-scp $BASE/dockerStartCommand.sh $CLOUDUSER@$PUBLICIP:
-ssh $CLOUDUSER@$PUBLICIP "chmod +x dockerStartCommand.sh"
-ssh $CLOUDUSER@$PUBLICIP "sudo ./dockerStartCommand.sh &"
+scp -i ./secrets/mdm_id_rsa $BASE/dockerStartCommand.sh $CLOUDUSER@$MDM_VM_IP:
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "chmod +x dockerStartCommand.sh"
+ssh -i ./secrets/mdm_id_rsa $CLOUDUSER@$MDM_VM_IP "sudo ./dockerStartCommand.sh &"
```

Comment:
Probably an out-of-scope point, but running sudo or making sudo available seems like a bad thing to be doing. I can see it was previously a part of the script before the change, so might have to be a discussion done elsewhere.